corehq/apps/app_manager/templatetags/xforms_extras.py

```diff
@@ -201,18 +207,17 @@ def _get_dynamic_input_trans_options(name, langs=None, allow_blank=True, is_text
                 options['placeholder'] = name[lang] if (allow_blank or name[lang] != '') else placeholder
                 options['lang'] = lang
             break
-    if is_textarea:
-        options = _escape_options(options, keys_to_html_escape=['value'])
-    else:
-        options = _escape_options(options)
+    options = _escape_options(options, escape_method=escape_method)
     return options
 
 
-def _escape_options(options, keys_to_html_escape: list[str] = []):
+def _escape_options(options, escape_method: EscapeMethod):
     escaped_options = {}
-    for (key, value) in options.items():
-        if key in keys_to_html_escape:
-            escaped_options[key] = html.escape(value)
-        else:
-            escaped_options[key] = html.escapejs(value)
+    if escape_method == EscapeMethod.HTML:
+        escape_func = html.escape
+    elif escape_method == EscapeMethod.JS:
+        escape_func = html.escapejs
+
+    for key, value in options.items():
+        escaped_options[key] = escape_func(value)
```

Comment:
nit: could also do this as a dict comprehension since there's no real logic applied:
```python
escaped_options = {key: escape_func(value) for key, value in options.items()}
```

Reply:
i like it [d682390](https://github.com/dimagi/commcare-hq/pull/35209/commits/d682390046b84c453ef6c7cb11112ee2769ffeab)

Reply:
Going this route I would also remove the `escaped_options = {}` initialization from line 215

Reply:
good point, thanks [ffc9027](https://github.com/dimagi/commcare-hq/pull/35209/commits/ffc9027a899ff7a0dddeb528be96c8ae283cbeb1)